exercises/practice/rotational-cipher/test_rotational_cipher.c

```diff
@@ -0,0 +1,100 @@
+#include "rotational_cipher.h"
+
+#include "test-framework/unity.h"
+
+#include <stdlib.h>
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void test_rotate(const char *expected, const char *text, int shift_key)
+{
+   char *rotated = rotate(text, shift_key);
+
+   TEST_ASSERT_EQUAL_STRING(expected, rotated);
+
+   free(rotated);
```

Comment:
Is it good to have this in the tests rather than the implementation?

It seems to imply a specific implementation or restrict the options for implementations.

Reply:
I'm not following. This exercise is about taking a string and returning a modified version of it. I think it's a standard pattern that the function allocates memory for the returned string and it's the caller's responsibility to free that memory. Similar exercises such as reverse-string, atbash-cipher, acronym, or pig-latin handle it in the exact same way.

Reply:
@ahans I think what @wolf99 means is that we should add a function in the implementation for freeing the result as we've done in other exercises (see `linked-list`, for example). I don't think that applies here since there's no freedom in the result type.